### PR TITLE
Add install_asm script

### PR DESCRIPTION
**Pull request by zerobfd**
This script aids in the installation of ASM.

### CODE REVIEW

**Review by elfinhe — APPROVED** [review]

**Comment by zerobfd** [comment]
Also, since we lost the commit history, I wanted to mention that this was a huge collaborative effort with @elfinhe @djazayeri @adammil2000 and @backward-compatible. Thank you all!

File: scripts/asm-installer/install_asm
Comment (jasonwzm) [inline scripts/asm-installer/install_asm:114]: `$((i+1))` here so that the output is 1 of 1 instead of 0 of 1.

File: scripts/asm-installer/install_asm
Comment (jasonwzm) [inline scripts/asm-installer/install_asm:414]: validate_gcp_resources needs to happen before configure_kubectl to make sure cluster and project exist.

**Review by jasonwzm — CHANGES_REQUESTED** [review]
Some small things found during initial run.

**Comment by elfinhe** [comment]
@jasonwzm mind taking a second round of review? I will wait for your approval before merging.

**Comment by yangminzhu** [comment]
should this be merged to release-1.6 branch as it's hard-coded to install ASM 1.6?

**Review by jasonwzm — CHANGES_REQUESTED** [review]

**Review by elfinhe — CHANGES_REQUESTED** [review]
Yangmin is right, this should be in release-1.6-asm branch.

**Comment by zerobfd** [comment]
Switching the base on the PR before changing the source seems to have messed it up, I'm going to close this one and re-open.

**Comment by zerobfd** [comment]
@jasonwzm's issue resolved + another bug at #142 . Closing this PR.

**Pull request closed without merging**
